fabric-events-player-v1/src/main/java/net/fabricmc/api/event/v1/player/PlayerRespawnCallback.java

```diff
@@ -0,0 +1,29 @@
+package net.fabricmc.api.event.v1.player;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.world.dimension.DimensionType;
+
+/**
+ * Represents a callback for when a {@link ServerPlayerEntity} is respawned.
+ *
+ * <p>This occurs when a player dies, or returns to the overworld from the end.</p>
+ */
+public interface PlayerRespawnCallback {
+	Event<PlayerRespawnCallback> EVENT = EventFactory.createArrayBacked(PlayerRespawnCallback.class, (callbacks) -> (newPlayer, oldPlayer, newDimension, died) -> {
+		for (PlayerRespawnCallback callback : callbacks) {
+			callback.onRespawn(newPlayer, oldPlayer, newDimension, died);
+		}
+	});
+
+	/**
+	 * Called when a player respawns.
+	 *
+	 * @param newPlayer The new {@link ServerPlayerEntity} that will be spawned.
+	 * @param oldPlayer The old {@link ServerPlayerEntity} that is being removed.
+	 * @param dimension The dimension this player is being respawned in.
+	 * @param alive Whether this player being respawned is still alive.
+	 */
+	void onRespawn(ServerPlayerEntity newPlayer, ServerPlayerEntity oldPlayer, DimensionType dimension, boolean alive);
```

Comment:
I would note that this is specifically whether `oldPlayer` is still alive, because `newPlayer` always is. 
Side note: I personally named this parameter `lossless`, as it defines whether the copy from `oldPlayer` to `newPlayer` should be completely lossless.

Reply:
I'll add that distinction to the javadoc when I have the chance.